search/reverse_geocoder.cpp

```diff
@@ -98,6 +117,44 @@ size_t ReverseGeocoder::GetMatchedStreetIndex(string const & keyName,
     }
   }
 
-  return (result < count ? result : streets.size());
+  return result;
 }
+
+void ReverseGeocoder::GetNearbyAddress(m2::PointD const & center,
+                                       Building & building, Street & street)
```

Comment:
How about to create a structure like Address with two fields (for now): building and street?